src/analyze/analyze-verify.c

```diff
@@ -218,13 +235,16 @@ static int verify_unit(Unit *u, bool check_man, const char *root) {
         return r;
 }
 
-int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run_generators, const char *root) {
+int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run_generators, RecursiveErrors recursive_errors, const char *root) {
         const ManagerTestRunFlags flags =
                 MANAGER_TEST_RUN_MINIMAL |
                 MANAGER_TEST_RUN_ENV_GENERATORS |
+                (recursive_errors == RECURSIVE_ERRORS_NO) * MANAGER_TEST_RUN_IGNORE_DEPENDENCIES |
                 run_generators * MANAGER_TEST_RUN_GENERATORS;
 
         _cleanup_(manager_freep) Manager *m = NULL;
+        _cleanup_set_free_free_ Set *s = NULL;
```

Comment:
hmm, this is no longer safe now that s can be set to POINTER_MAX. You need to define your own little stub destructor for this:

```c
static void set_destroy_ignore_pointer_max(Set** s) {
        if (*s == POINTER_MAX)
                 return;
        set_free_free(*s);
}
```

Reply:
what about doing this check in set_free itself? POINTER_MAX is never valid, no?

Reply:
i thought about that, but this would open quite a can of worms, because it would call for us to do that for *all* destructors really, and we have *so* *so* many of those.

Reply:
ok